src/golang/config/config.go

```diff
@@ -45,8 +45,8 @@ func APIKey() string {
 }
 
 // Storage returns the storage layer config.
-func Storage() shared.StorageConfig {
-	return *globalConfig.StorageConfig
+func Storage() *shared.StorageConfig {
```

Comment:
@saurav-c I found making it returning a pointer is more convenient for other parts of the codebase. Just making sure you are aware of this and there's no side effect.

Reply:
It's more convenient, but do you think it's more dangerous to return a pointer, because then the caller needs to make sure not to change any of the struct values and since this is a globally shared object this could result in weird side effects.

Reply:
I see, yeah I think it make sense to be safe. I'll revert the change.